### PR TITLE
Add UT test for nightly

### PR DESCRIPTION
**Pull request by RUIJIEZHONG66166**

### CODE REVIEW

File: .github/workflows/inductor_xpu_ut_nightly.yml
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_ut_nightly.yml:1]: Change this callable ut workflow name to `_linux_ut.yml`, and use it in `pull.yml` directly

File: .github/workflows/inductor_xpu_ut_nightly.yml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      torch_xpu_ops_update:
+        required: false
+        type: string
+        default: 'true'
+        description: True means update xpu_ops when building pytorch, otherwise means not
+      expected_fail_num:
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_ut_nightly.yml:11]: I think we don't need this parameter for UT test. Please add a parameter for UT test scope, default value include example, extend UT, xpu op UT

File: .github/workflows/inductor_xpu_ut_nightly.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          pip install -r requirements.txt
+          export USE_XPU=1
+          source /opt/intel/oneapi/compiler/latest/env/vars.sh
+          export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
+          python setup.py bdist_wheel
+          pip install --force-reinstall dist/*.whl
+          git clone https://github.com/pytorch/vision && cd vision && python setup.py install && cd ..
+          pip install -r .ci/docker/requirements-ci.txt
+      - name: Run XPU OP Examples
+        if: ${{ hashFiles('examples/') != '' }}
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_ut_nightly.yml:58]: Please also add ut  test scope condition for each kind of UT test

File: .github/workflows/inductor_xpu_ut_nightly.yml [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            echo -e "[ERROR] Inductor UT test for XPU OP UT failed number > ${{ inputs.expected_fail_num }}"
+            exit 1
+          fi
+
+          if if [[ $num_failed_torch_xpu -gt ${{ inputs.expected_fail_num }} ]] && [[ $num_passed_torch_xpu -lt 0 ]]; then
+            echo -e "[ERROR] Inductor UT test for Torch XPU UT failed number > ${{ inputs.expected_fail_num }}"
+            exit 1
+          fi
+          
+      - name: Upload Inductor XPU UT Data
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_ut_nightly.yml:199]: Add `if: always()`

File: .github/workflows/inductor_xpu_e2e_nightly.yml [hunk 4]
```diff
@@ -244,6 +254,13 @@ jobs:
           name: Inductor-XPU-E2E-Data-${{ github.event.pull_request.number || github.sha }}
           path: ${{ github.workspace }}/upload_files
 
+  Inductor-XPU-UT-Nightly-Tests:
+    if: inputs.test_ut == 'true'
+    name: Nightly Inductor XPU UT Test
+    uses: ./.github/workflows/inductor_xpu_ut_nightly.yml
+    with:
+      expected_fail_num: 0
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_e2e_nightly.yml:262]: don't need this one

File: .github/workflows/inductor_xpu_e2e_nightly.yml [hunk 5]
```diff
@@ -41,6 +41,16 @@ on:
...
+      torch_xpu_ops_update:
+        required: false
+        type: string
+        default: 'true'
+        description: True means update xpu_ops when building pytorch, otherwise means not
+      test_ut:
+        required: true
+        type: string
+        default: 'true'
+        description: True will running UT test for Ondemand test, false will not.
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_e2e_nightly.yml:54]: Can we also add pytorch branch in here also?

File: .github/workflows/inductor_xpu_ut_nightly.yml [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        default: 'true'
+        description: True means update xpu_ops when building pytorch, otherwise means not
+      expected_fail_num:
+        required: false
+        type: number
+        description: for UT result check
+
+jobs:
+  Inductor-XPU-UT-Tests:
+    runs-on: linux.idc.xpu
```
Comment (chuanqi129) [inline .github/workflows/inductor_xpu_ut_nightly.yml:18]: please also add this runner label as this callable workflow inputs, default `linux.idc.xpu`

File: .github/workflows/inductor_xpu_ut_nightly.yml (hunk 2 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_ut_nightly.yml]: Done

File: .github/workflows/inductor_xpu_ut_nightly.yml (hunk 3 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_ut_nightly.yml]: Done

File: .github/workflows/inductor_xpu_e2e_nightly.yml (hunk 4 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_e2e_nightly.yml]: Done

File: .github/workflows/inductor_xpu_ut_nightly.yml (hunk 6 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_ut_nightly.yml]: Done

File: .github/workflows/inductor_xpu_e2e_nightly.yml (hunk 5 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_e2e_nightly.yml:59]: Done

File: .github/workflows/inductor_xpu_ut_nightly.yml (hunk 1 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_ut_nightly.yml]: Done

File: .github/workflows/inductor_xpu_ut_nightly.yml
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_ut_nightly.yml:1]: Done

**Review by chuanqi129 — APPROVED** [review]

**Pull request merged**